src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -72,6 +79,45 @@ private WebAssemblyHostBuilder()
         /// </summary>
         public IServiceCollection Services { get; }
 
+        /// <summary>
+        /// Registers a <see cref="IServiceProviderFactory{TBuilder}" /> instance to be used to create the <see cref="IServiceProvider" />.
+        /// </summary>
+        /// <param name="factory">The <see cref="IServiceProviderFactory{TBuilder}" />.</param>
+        /// <typeparam name="TBuilder">The type of builder provided by the <see cref="IServiceProviderFactory{TBuilder}" />.</typeparam>
+        public void ConfigureContainer<TBuilder>(IServiceProviderFactory<TBuilder> factory)
+        {
+            if (factory == null)
+            {
+                throw new ArgumentNullException(nameof(factory));
+            }
+
+            ConfigureContainer(factory, configure: null);
+        }
+
+        /// <summary>
+        /// Registers a <see cref="IServiceProviderFactory{TBuilder}" /> instance to be used to create the <see cref="IServiceProvider" />.
+        /// </summary>
+        /// <param name="factory">The <see cref="IServiceProviderFactory{TBuilder}" />.</param>
+        /// <param name="configure">
+        /// A delegate used to configure the <typeparamref T="TBuilder" />. This can be used to configure services using
+        /// APIS specific to the <see cref="IServiceProviderFactory{TBuilder}" /> implementation.
```

Comment:
```suggestion
        /// APIs specific to the <see cref="IServiceProviderFactory{TBuilder}" /> implementation.
```

Reply:
TBH I don't really get why `configure` is required, since any configuration logic you want to run could instead be put inside either the `CreateBuilder` or `CreateServiceProvider` methods on your implementation.

Is this because there are pre-existing API patterns in 3rd-party DI libraries that assume this call flow?

Reply:
This is because the app developer doesn't write `IServiceProviderFactory<>` - it's part of AutoFac or some other package they pulled in. `Action<TBuilder>` gives them a chance to run code against that factory's specific builder API.